modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/GenericAsyncAggregateCommandModel.scala

```diff
@@ -0,0 +1,56 @@
+// Copyright © 2017-2021 UKG Inc. <https://www.ukg.com>
+
+package com.ukg.surge.multilanguage
+
+import akka.actor.ActorSystem
+import akka.event.{ Logging, LoggingAdapter }
+import com.ukg.surge.multilanguage
+import com.ukg.surge.multilanguage.protobuf._
+import surge.scaladsl.command.AsyncAggregateCommandModel
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+class GenericAsyncAggregateCommandModel(bridgeToBusinessApp: BusinessLogicService)(implicit system: ActorSystem)
+    extends AsyncAggregateCommandModel[SurgeState, SurgeCmd, SurgeEvent] {
+
+  import Implicits._
+  val logger: LoggingAdapter = Logging(system, classOf[GenericAsyncAggregateCommandModel])
+  import system.dispatcher
+
+  override def executionContext: ExecutionContext = system.dispatcher
+
+  override def processCommand(aggregate: Option[SurgeState], surgeCommand: SurgeCmd): Future[Seq[SurgeEvent]] = {
+    logger.info(s"""Calling command handler of business app via gRPC.
+         |Aggregate id: ${surgeCommand.aggregateId}).
+         |State defined: ${aggregate.isDefined}.
+         |Command payload size: ${surgeCommand.payload.length} (bytes).""".stripMargin)
+
+    val maybePbState: Option[protobuf.State] = aggregate.map(surgeState => surgeState: protobuf.State)
+    val pbCommand: protobuf.Command = surgeCommand: multilanguage.protobuf.Command
+    val processCommandRequest = ProcessCommandRequest(maybePbState, Some(pbCommand))
+    val reply: Future[ProcessCommandReply] = bridgeToBusinessApp.processCommand(processCommandRequest)
+
+    reply.flatMap { processCommandReply =>
+      {
+        if (processCommandReply.isSuccess) {
+          Future.successful {
+            processCommandReply.events.map(e => e: SurgeEvent)
+          }
+        } else {
+          Future.failed(new Exception(processCommandReply.rejectionMessage))
```

Comment:
One potential place for some weirdness I could see would be where business logic to process commands throws an exception or returns a failed future explicitly rather than rejecting with a simple message. Maybe the SDK can handle the wrapping of that somehow. Just thoughts for the future - this is good enough for now to at least prove out that it works